src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -71,13 +71,25 @@ type uvi =
     | TERM of ctx_uvar * term
     | UNIV of S.universe_uvar * universe
 
+type defer_ok_t = 
+  | NoDefer
+  | DeferAny
+  | DeferFlexFlexOnly
```

Comment:
defer_ok used to be a boolean. I added a third case, DeferFlexFlexOnly, which allows deferring only flex-flex constraints